File: add-analytics.py
#!/usr/bin/env python3
import os
import re

analytics_script = '''<!-- Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=G-V45PK6W060"></script>
<script>
window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'G-V45PK6W060');
</script>'''

blog_dir = "/home/jquijanoq/.openclaw/workspace/bnwraptor-blog"
updated = []
skipped = []

for filename in os.listdir(blog_dir):
    if not filename.endswith('.html'):
        continue
    if filename == 'drafts':
        continue
    filepath = os.path.join(blog_dir, filename)
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()
    
    if 'G-V45PK6W060' in content:
        skipped.append(filename)
        continue
    
    if '</head>' in content:
        new_content = content.replace('</head>', analytics_script + '</head>')
        with open(filepath, 'w', encoding='utf-8') as f:
            f.write(new_content)
        updated.append(filename)
    else:
        print(f"No </head> found in: {filename}")

print(f"\nUpdated: {len(updated)}")
for f in updated:
    print(f"  + {f}")
print(f"\nSkipped (already has GA): {len(skipped)}")
for f in skipped:
    print(f"  = {f}")

File: add-analytics.js
const fs = require('fs');
const path = require('path');

const analyticsScript = `<!-- Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=G-V45PK6W060"></script>
<script>
window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'G-V45PK6W060');
</script>`;

const blogDir = __dirname;
const updated = [];
const skipped = [];

fs.readdirSync(blogDir).forEach(filename => {
  if (!filename.endsWith('.html')) return;
  if (filename === 'drafts' || filename === 'add-analytics.py') return;

  const filepath = path.join(blogDir, filename);
  let content = fs.readFileSync(filepath, 'utf8');

  if (content.includes('G-V45PK6W060')) {
    skipped.push(filename);
    return;
  }

  if (content.includes('</head>')) {
    content = content.replace('</head>', analyticsScript + '</head>');
    fs.writeFileSync(filepath, content);
    updated.push(filename);
  } else {
    console.log(`No </head> in: ${filename}`);
  }
});

console.log(`\nUpdated: ${updated.length}`);
updated.forEach(f => console.log(`  + ${f}`));
console.log(`\nSkipped: ${skipped.length}`);
skipped.forEach(f => console.log(`  = ${f}`));